src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,300 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Feature extractor class for Pop2Piano"""
+
+import warnings
+from typing import List, Optional, Union
+
+import librosa
+import numpy as np
+import scipy
+import torch
+from torch.nn.utils.rnn import pad_sequence
+
+from ...audio_utils import fram_wave, get_mel_filter_banks, stft
+from ...feature_extraction_sequence_utils import SequenceFeatureExtractor
+from ...feature_extraction_utils import BatchFeature
+from ...utils import OptionalDependencyNotAvailable, TensorType, is_essentia_available, logging
+
+
+try:
+    if not is_essentia_available:
+        raise OptionalDependencyNotAvailable()
+except ImportError:
+    raise ImportError("There was an error while importing essentia!")
+else:
+    import essentia
+    import essentia.standard
+
+
+logger = logging.get_logger(__name__)
+
+
+class Pop2PianoFeatureExtractor(SequenceFeatureExtractor):
+    r"""
+    Constructs a Pop2Piano feature extractor.
+
+    This feature extractor inherits from [`~feature_extraction_sequence_utils.SequenceFeatureExtractor`] which contains
+    most of the main methods. Users should refer to this superclass for more information regarding those methods.
+
+    Args:
+    This class extracts rhythm and does preprocesses before being passed through the transformer model.
+        sampling_rate (`int`, *optional*, defaults to 22050):
+            Sample rate of audio signal.
+        use_mel (`bool`, *optional*, defaults to `True`):
```

Comment:
As far as I understand, Pop2Piano **only** works with log-mel spectrogram inputs? E.g. Figure 2 from the [paper](https://arxiv.org/pdf/2211.00895.pdf). If this is the case, let's remove the argument `use_mel` and always just compute the spectrogram